_posts/en/posts/2016-06-06-compact-blocks-faq.md

```diff
@@ -0,0 +1,144 @@
+
+When running live experiments in ‘high bandwidth’ mode and having nodes send up to 6 transactions they predict their peer doesn’t have, we can expect to see around 86% of blocks propagate immediately without needing to request any missing transactions. Even without sending any transactions that are predicted to be missing, a little bit less than 50% of blocks propagate immediately, with the rest needing a full additional network round trip to request and receive missing transactions.
+
+Since the difference between mempools and blocks is fewer than 6 transactions in most cases, this means that compact block relay achieves a dramatic reduction in required peak bandwidth.
+
+## How are expected missing transactions chosen to immediately forward?
+
+To reduce the number of things that need to be reviewed in the initial implementation, only the coinbase transaction will be pre-emptively sent.
+
+However, in the described experiments, the sending node used a simple formula to choose which transactions to send: when Node A received a block, it checked to see which transactions were in the block but not in its mempool; those were the transactions it predicted that its peer didn't have.  The reasoning is that (without additional information) the transactions you didn't know about are probably also the transactions your peers don't know about. With this basic heuristic, a large improvement was seen, illustrating that many times the simplest solutions are the best.
+
+##How does the Fast Relay Network factor into this?
+
+The [Fast Relay Network](http://bitcoinrelaynetwork.org/) (FRN) consists of two pieces:
+
+* The curated set of nodes currently in the Fast Relay Network
+
+* The Fast Block Relay Protocol (FBRP)
+
+The set of curated nodes in the FRN have been carefully chosen with minimal relay over the globe as the number one priority. Failure of these nodes would result in a not inconsequential increase of wasted hashpower and potential further centralization of mining. A large majority of mining hashpower today connects to this network.
+
+The FBRP is how the participating nodes communicate block information to each other. Nodes keep track of what transactions they send to each other, and relay block differentials based off of this knowledge. This protocol is nearly optimal for one-to-one server-client communication of new blocks. This however requires a not-well connected relay topology is more brittle than a more general p2p network. Improvements at the protocol level using compact blocks will shrink the performance gap between the curated network of nodes and the p2p network in general. The increased robustness of the p2p network and block propagation speed at large will play a role in how the network develops in the future. 
```

Comment:
"Nodes keep track of what transactions they send to each other, and relay block differentials based off of this knowledge. This protocol is nearly optimal for one-to-one server-client communication of new blocks"

No longer true. Now uses UDP with FEC and rumoring among each other in a P2P-ish kind of way.